<launch>

  <!--Namespace of ceiling camera topics and frames -->
  <arg name="cameraPrefix" default="ceiling" />

  <!--Image topic. The default image topic. Just an image topic.  -->
  <arg name="image_topic" default="$(arg cameraPrefix)/rgb/image_color"/>

  <!-- Camera info topic. -->
  <arg name="camera_topic" default="$(arg cameraPrefix)/rgb/camera_info"/>

  <node name="ceilingLocalizer" pkg="privacy" type="ceilingLocalizer.py">
    <param name="image_topic" value="$(arg image_topic)"/>
    <param name="camera_topic" value="$(arg camera_topic)"/>
  </node>

  <!-- Publishes the transform between marker and base_footprint. -->
  <node name="base_publisher" pkg="tf" type="static_transform_publisher" args="0 0 0.42 1 0.0001 0 0 base_top base_footprint 100"></node>

  <!-- Publishes the transform between ceiling camera and map. -->  
  <node name="map_publisher" pkg="tf" type="static_transform_publisher" args="0 0 2.709 0 1.57 0 map ceiling_link 100"></node>

</launch>
